Symfony/CS/Fixer/Contrib/MultipleUnsetFixer.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Token;
+use Symfony\CS\Tokenizer\Tokens;
+
+class MultipleUnsetFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+
+        // Offset start at 5, because any call to 'unset' that can possible be merged is
+        // build up out of at least 4 Tokens; 'unset', '(', '[var]', ')' after at least one ';'.
+        for ($index = $tokens->count() - 6; $index > 5; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_UNSET)) {
+                continue;
+            }
+
+            $previousUnsetCall = $this->getPreviousUnsetCall($tokens, $index);
+            if (is_int($previousUnsetCall)) {
+                $index = $previousUnsetCall;
+                continue;
+            }
+
+            list($previousUnset, $previousUnsetBraceStart, $previousUnsetBraceEnd, $previousUnsetSemicolon) = $previousUnsetCall;
+
+            // Merge the tokens inside the 'unset' call into the previous one 'unset' call.
+            $tokensAddCount = $this->moveTokens(
+                $tokens,
+                $nextUnsetContentStart = $tokens->getNextTokenOfKind($index, array('(')),
+                $nextUnsetContentEnd = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $nextUnsetContentStart),
+                $previousUnsetBraceEnd - 1
+            );
+
+            $tokens->insertAt($previousUnsetBraceEnd, new Token(','));
+            ++$tokensAddCount;
+
+            // Remove 'unset', '(', ')' and (possibly) ';' from the merged 'unset' call.
+            $this->clearOffsetTokens($tokens, $tokensAddCount, array($index, $nextUnsetContentStart, $nextUnsetContentEnd));
+
+            $nextUnsetSemicolon = $tokens->getNextMeaningfulToken($nextUnsetContentEnd);
+            if (null !== $nextUnsetSemicolon && $tokens[$nextUnsetSemicolon]->equals(';')) {
+                $tokens[$nextUnsetSemicolon]->clear();
+                $highestClearTokenIndex = $nextUnsetSemicolon;
+            } else {
+                $highestClearTokenIndex = $tokensAddCount + $nextUnsetContentEnd;
+            }
+
+            // Since tokens are cleared and inserted there might be successive whitespace tokens which need to
+            // be merge into one whitespace token. These tokens can only be after the first 'unset' call ';' token
+            // up to the token after the last cleared token of the second 'unset' + 1
+            $this->mergeWhitespaceTokens(
```

Comment:
It is  the 2nd (+?) place of that kind of logic, we should extract it to one place.